### PR TITLE
Update va-select events

### PR DESCRIPTION
**Pull request by andresriveratoro**
## Chromatic

https://671-va-select-event--60f9b557105290003b387cd5.chromatic.com

## Description
Closes https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/671

## Testing done
E2E, Storybook

## Screenshots
Screen Shot 2022-04-07 at 14 03 23

## Acceptance criteria
- [x] Replace `vaKeyDown` with `keydown` native event
- [x] Update `vaSelect` event propagation

## Definition of done
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

**Review by bkjohnson — COMMENTED** [review]
Do we know if the team is having an issue with the keydown event or the change one?

**Comment by andresriveratoro** [comment]
> Do we know if the team is having an issue with the keydown event or the change one?

I believe it's the `change` event. I don't see any teams using the `keydown` event.

File: packages/web-components/src/components/va-select/va-select.tsx [hunk 1]
```diff
@@ -48,15 +48,14 @@ export class VaSelect {
...
-
   /**
    * The event emitted when the selected value changes
    */
-  @Event() vaSelect: EventEmitter;
+  @Event({
+    bubbles: true,
+    composed: true,
+  })
+  vaSelect: EventEmitter;
```
Comment (bkjohnson) [inline packages/web-components/src/components/va-select/va-select.tsx:58]: Oh, this is what should make that test easier for the app team, right? I was too focused on the `keydown` changes to notice this the first time :upside_down_face:

File: packages/storybook/stories/va-select.stories.jsx [hunk 2]
```diff
@@ -79,6 +91,7 @@ const Template = ({
         error={error}
         aria-live-region-text={ariaLiveRegionText}
         use-add-button={useAddButton}
+        onKeyDown={() => console.log('KEYDOWN FIRED')}
```
Comment (bkjohnson) [inline packages/storybook/stories/va-select.stories.jsx:94]: I'm also in favor of this native event + handler replacing the custom one :+1:

File: packages/web-components/src/components/va-select/va-select.tsx (hunk 1 above)
Comment (andresriveratoro) [reply packages/web-components/src/components/va-select/va-select.tsx:58]: The tests are still failing locally when using ```userEvent.click(await screen.findByText(/use your current location/i));```. 😞

**Comment by andresriveratoro** [comment]
I have been unable to observe the `vaChange` event fire in `vets-website` tests. 

This code allows you to select a new value using `userEvent` and the test passes, but `vaChange` isn't fired.
```jsx
const select = await screen.findByTestId('providersSelect');
userEvent.selectOptions(select, 'Two');
expect(screen.getByRole('option', { name: 'Two' }).selected).to.equal(true);
```

This throws a `Error: The given element does not have a value setter`
```jsx
fireEvent.change(select, { target: { value: 'Two' } });
```

#### Related Issues
https://github.com/testing-library/dom-testing-library/issues/413
https://github.com/testing-library/dom-testing-library/issues/742

**Comment by pdavies88** [comment]
@andresriveratoro can we notate your last comment on the ticket: https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/671 and tag Paul so he's aware

**Review by pdavies88 — APPROVED** [review]

**Pull request closed without merging**